manual/source/classes/contributed_definitions/cgms-structure.rst

```diff
@@ -0,0 +1,286 @@
+
+    :ref:`NXcg_marching_cubes`:
+        An approach to store metadata of a specific implementation of
+        the Marching Cubes algorithm, whose sensitivity to specific topological
+        configurations is known to result in different triangle soups.
+
+    :ref:`NXdelocalization`:
+        An approach to document procedures whereby a scalar field
+        is smoothened in a controlled manner.
+
+    :ref:`NXsimilarity_grouping`:
+        An alias for NXclustering.
+
+    :ref:`NXclustering`:
+        A description for clustering of objects (such as atoms or features).
+
+    :ref:`NXorientation_set`:
+        A set of rotations to describe the relative orientation of members of a set of features/objects.
+
+    :ref:`NXslip_system_set`:
+        Metadata to a set of slip system/slip system family for
+        a given crystal structure.
+
+    :ref:`NXms_feature_set`:
+        Generic base class to describe any nested set of features
+        of a microstructure at the continuum-, micron-, nano-scale, or
+        to represent a topology of molecules and atoms.
+
+    :ref:`NXms_snapshot`:
+        A container to describe the state of microstructural features
+        at a given point in time.
+
+    :ref:`NXms_snapshot_set`:
+        The corresponding class to hold a set of :ref:`NXms_snapshot` objects.
+
+    :ref:`NXchemical_composition`:
+        (Chemical) composition of a sample or a set of things.
+
+Furthermore, it can be useful to have a set of base classes with
+which software documents it state and gives a summary for users about the performance
+and elapsed time measured while processing data. These utility classes include:
+
+    :ref:`NXprogram`:
+        A named and version of a program of library/component.
+
+    :ref:`NXcs_filter_boolean_mask`:
+        A boolean mask.
+
+    :ref:`NXcs_prng`:
+        Metadata of a pseudo-random number generator (PRNG) algorithm.
+
+    :ref:`NXcs_profiling`:
+        A structuring group holding a set of :ref:`NXcs_profiling_event` instances.
+
+    :ref:`NXcs_profiling_event`:
+        Profiling/benchmark data to an event of
+        tracking an algorithm/computational step.
+
+    :ref:`NXcs_computer`:
+        Metadata of a computer.
+
+    :ref:`NXcs_cpu`:
+        Metadata of a central processing unit.
+
+    :ref:`NXcs_gpu`:
+        Metadata of a graphical processing unit / accelerator.
+
+    :ref:`NXcs_mm_sys`:
+        Metadata of the (main) memory (sub-)system.
+
+    :ref:`NXcs_io_sys`:
+        Metadata of the input/output system.
+
+    :ref:`NXcs_io_obj`:
+        Metadata of a component storing data of an :ref:`NXcs_io_sys` instance.
+
+.. _IcmeMsModels:
+
+Application definitions for ICME models
+#######################################
+
+It is important to embrace the large research community of materials engineers
+as they are frequent users of electron microscopy and atom probe microscopy.
+In this community frequently ICME microstructure models are used. ICME is an
```

Comment:
"... ICME (Integrated Computational Materials Engineering) microstructure models are used. These models are derived from a design ..."

Reply:
moved to an own section icme-structure

Reply:
Edited in PR1271 now without being moved to an own icme-structure.rst